clusterctl/clusterdeployer/clusterclient.go

```diff
@@ -65,15 +67,16 @@ func (c *clusterClient) removeKubeconfigFile() error {
 	return os.Remove(c.kubeconfigFile)
 }
 
-func NewClusterClientFromFile(kubeconfigFile string) (*clusterClient, error) {
-	c, err := clientcmd.NewClusterApiClientForDefaultSearchPath(kubeconfigFile)
+func NewClusterClientFromDefaultSearchPath(kubeconfigFile string, overrides tcmd.ConfigOverrides) (*clusterClient, error) {
```

Comment:
I'm not sure I understand the name of this fn; why not name it NewFromFile(...)?